recipe/meta.yaml

```
@@ -1,88 +1,108 @@
-{% set version = "4.2.2" %}
+{% set name = "ffmpeg" %}
+{% set version = "6.1.1" %}
+# win specific vars
+{% set version_majmin = '.'.join(version.split('.', 2)[:2]) %}
+{% set release_date = "2023-12-31" %}
+{% set release_time = "12-55" %}
 
 package:
-  name: ffmpeg
+  name: {{ name }}
   version: {{ version }}
 
 source:
-  - url: https://ffmpeg.org/releases/ffmpeg-{{ version }}.tar.gz  # [not win]
-    sha256: 83f9a9aa0acf8036daf47494d99a8c31154a18ebb6841d89878ba47783559bd0  # [not win]
-
-  - url: https://ffmpeg.zeranoe.com/builds/win64/shared/ffmpeg-{{ version }}-win64-shared.zip  # [win]
-    sha256: 5aedf268952b7d9f6541dbfcb47cd86a7e7881a3b7ba482fd3bc4ca33bda7bf5  # [win]
-
-  - url: https://ffmpeg.zeranoe.com/builds/win64/dev/ffmpeg-{{ version }}-win64-dev.zip  # [win]
-    sha256: f4885f859c5b0d6663c2a0a4c1cf035b1c60b146402790b796bd3ad84f4f3ca2  # [win]
+  url: https://github.com/FFmpeg/FFmpeg/archive/refs/tags/n{{ version }}.tar.gz
+  sha256: 7c1ebea95d815e49c1e60c7ee816410dec73a81b8ac002b276780d2f9048e598
+  patches:
+    - patches/0001-use-include-path-of-our-libxml2.patch # [win]
 
 build:
   number: 0
-  # The windows build is repacking binaries rather than building from source
-  run_exports:
-  # seems to be minor version compatibility
-  # https://abi-laboratory.pro/tracker/timeline/ffmpeg/
-    - {{ pin_subpackage('ffmpeg', max_pin='x.x') }}
-  missing_dso_whitelist:  # [osx]
-    - /System/Library/Frameworks/VideoToolbox.framework/Versions/A/VideoToolbox  # [osx]
+  ignore_run_exports:
+    - libcxx
+    # these seems to be overdepending only on linux
+    - libstdcxx-ng  # [linux]
+    # this seems to be statically linked only on osx
 
 requirements:
   build:
     - {{ compiler("c") }}
     - {{ compiler("cxx") }}
-    - pkg-config  # [not win]
+    - autotools_clang_conda  # [win]
+    - pkg-config
     - libtool  # [not win]
-    - nasm  # [(osx and x86_64) or linux32 or linux64]
+    - nasm  # [win or (osx and x86_64) or (linux and (not s390x))]
     - make  # [not win]
+    - patch  # [not win]
+    - m2-patch  # [win]
   host:
-    - bzip2  # [not win]
-    - freetype  # [not win]
-    - gnutls  # [not win]
-    - libiconv  # [not win and not linux]
-    - x264  # [not win]
-    - zlib  # [not win]
+    - aom
+    - bzip2
+    - dav1d
+    - freetype
+    - fontconfig
+    - harfbuzz
+    - libiconv
+    - zlib
     - openh264  # [not win]
+    - openjpeg >=2.1.0   # [not win]
+    - librsvg  # [osx]
+    - libtheora
+    - libvorbis # [not win]
+    - libxml2
```

```suggestion
    - libxml2 {{ libxml2 }}
```
It will help us with `libxml2` rebuildings. I guess other `host` libraries from `cbc.yaml` should be pinned in the same manner.

---

done